src/use-cases/authenticate.ts

```diff
@@ -20,100 +19,99 @@ import { UserLoggedInPayload, LiberoEventType } from '@libero/event-types';
-                            token_id: v4(),
-                            token_version: '0.1-alpha',
-                            identity,
-                            roles: [{ journal: 'elife', kind: 'author' }],
-                            meta: null,
-                        };
-
-                        if (emailAddress) {
-                            payload.meta = {
-                                email: emailAddress,
-                            };
-                        }
-
-                        const outputToken = encode(payload);
-
-                        // send audit logged in message
-                        const auditEvent: Event<UserLoggedInPayload> = {
-                            id: v4(),
-                            created: new Date(),
-                            eventType: LiberoEventType.userLoggedInIdentifier,
-                            context: {
-                                source: 'continuum-auth',
-                            },
-                            payload: {
-                                name: profile.name.preferred,
-                                userId: payload.identity.user_id,
-                                email: profile.emailAddresses.length > 0 ? profile.emailAddresses[0].value : '',
-                                result: 'authorized',
-                                timestamp: new Date(),
-                            },
-                        };
-                        eventBus.publish(auditEvent);
-
-                        res.redirect(`${login_return_url}#${outputToken}`);
-                    })
-                    .getOrElseL(() => {
-                        logger.warn('unauthorized');
-                        res.status(403).json({ ok: false, msg: 'unauthorised' });
-                    });
+): Promise<void> => {
+    if (!('token' in req.params)) {
+        logger.warn('noTokenProvided');
+        res.status(500).json({ ok: false });
+        return;
+    }
+
+    const token = req.params['token'];
+    // Decode the token that's passed to this endpoint from whatever OAuth provider we go with (I'm guessing ORCiD)
+    // Somehow resolve that user's identifier/metadata from the profiles service
+    // Shove a subset of that information into a JWT
+    // Send that back to the client
+
+    // Controller: perform the requests to the various services and fetch the user data
+
+    const {
+        auth: { login_return_url },
+    } = config;
+
+    const parsedToken = decodeJournalToken(token);
+
+    if (parsedToken.isEmpty()) {
+        logger.error('Invalid token');
+        res.status(403).json({ ok: false, msg: 'unauthorised' });
+        throw new Error();
+    }
+
+    const id = parsedToken.get();
+    const maybeProfile = await profilesService.getProfileById(id);
+
+    if (maybeProfile.isEmpty()) {
+        logger.warn('unauthorized');
+        res.status(403).json({ ok: false, msg: 'unauthorised' });
```

Comment:
Should we be updating the return `msg` as part of this to something like `profile not found` ?

Reply:
also the `logger.warn` might benefit from some more specific info?